matrix_sdk_base/src/client.rs

```diff
@@ -959,6 +980,59 @@ impl BaseClient {
         }
     }
 
+    /// Receive a successful /messages response.
+    ///
+    /// * `response` - The successful response from /messages.
+    pub async fn receive_messages(
+        &self,
+        room_id: &RoomId,
+        direction: &Direction,
+        response: &GetMessageEventsResponse,
+    ) -> Result<Vec<SyncRoomEvent>> {
+        let mut changes = StateChanges::default();
+
+        let mut events: Vec<SyncRoomEvent> = vec![];
+        for event in &response.chunk {
+            #[allow(unused_mut)]
+            let mut event: SyncRoomEvent = event.clone().into();
+
+            #[cfg(feature = "encryption")]
+            match hoist_room_event_prev_content(&event.event) {
+                Ok(AnySyncRoomEvent::Message(AnySyncMessageEvent::RoomEncrypted(encrypted))) => {
+                    if let Some(olm) = self.olm_machine().await {
+                        if let Ok(decrypted) = olm.decrypt_room_event(&encrypted, room_id).await {
+                            event = decrypted;
+                        }
+                    }
+                }
+                Ok(_) => {}
+                Err(error) => {
+                    warn!("Error deserializing event {:?}", error);
+                }
+            }
+
+            events.push(event);
+        }
+
+        let (chunk, start, end) = match direction {
+            Direction::Backward => {
+                (events.clone(), response.start.clone().unwrap(), response.end.clone())
```

Comment:
The unwrap either need to be gone or, if we're sure that it's ok, turned into an except.